pallets/parachain-staking/src/vote.rs

```diff
@@ -0,0 +1,223 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(not(feature = "std"))]
+extern crate alloc;
+#[cfg(not(feature = "std"))]
+use alloc::string::String;
+
+use sp_std::marker::PhantomData;
+use codec::{Decode, Encode, MaxEncodedLen};
+use sp_avn_common::{
+    event_types::Validator,
+    offchain_worker_storage_lock::{self as OcwLock},
+};
+use sp_std::prelude::*;
+
+use frame_support::{
+    dispatch::{DispatchError, DispatchResult},
+    log,
+};
+use frame_system::offchain::SubmitTransaction;
+use pallet_avn::{self as avn, vote::*, Error as avn_error};
+use sp_application_crypto::RuntimeAppPublic;
+use sp_core::ecdsa;
+use sp_runtime::scale_info::TypeInfo;
+use sp_std::fmt::Debug;
+
+use super::{Call, Config};
+use crate::{Pallet as ParachainStaking, GrowthId, Store, AVN, GrowthPeriodIndex};
+
+pub const CAST_VOTE_CONTEXT: &'static [u8] = b"growth_casting_vote";
+pub const END_VOTING_PERIOD_CONTEXT: &'static [u8] = b"growth_end_voting_period";
+const MAX_VOTING_SESSIONS_RETURNED: usize = 5;
+
+#[derive(PartialEq, Eq, Clone, Encode, Decode, Default, Debug, MaxEncodedLen, TypeInfo)]
+pub struct GrowthVotingSession<T: Config> {
+    growth_id: GrowthId,
+    phantom: PhantomData<T>
+}
+
+impl<T: Config> GrowthVotingSession<T> {
+    pub fn new(growth_id: &GrowthId) -> Self {
+        return GrowthVotingSession { growth_id: growth_id.clone(), phantom: Default::default() }
+    }
+}
+
+impl<T: Config> VotingSessionManager<T::AccountId, T::BlockNumber> for GrowthVotingSession<T> {
+    fn cast_vote_context(&self) -> &'static [u8] {
+        return CAST_VOTE_CONTEXT
+    }
+
+    fn end_voting_period_context(&self) -> &'static [u8] {
+        return END_VOTING_PERIOD_CONTEXT
+    }
```

Comment:
do we need to have functions merely to access a constant? Can we not  use the constant directly?

Reply:
These functions are interface functions that need to be implemented for each pallet.